pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -324,37 +324,12 @@ private boolean registerNamespace(String namespace, boolean ensureOwned) throws
             targetMap = findingBundlesNotAuthoritative;
         }
 
-        return targetMap.computeIfAbsent(bundle, (k) -> {
-            CompletableFuture<Optional<LookupResult>> future = new CompletableFuture<>();
-
-            // First check if we or someone else already owns the bundle
-            ownershipCache.getOwnerAsync(bundle).thenAccept(nsData -> {
-                if (!nsData.isPresent()) {
-                    // No one owns this bundle
+        if (targetMap.get(bundle) != null && !targetMap.get(bundle).isDone()) {
+            return findBrokerServiceUrlInternal(bundle, authoritative, readOnly);
```

Comment:
shouldn't we return the `future` here, since there is already an ongoing lookup, no?